src/main/java/com/enonic/lib/guillotine/macro/ContentLinkProcessorManager.java

```diff
@@ -0,0 +1,297 @@
+    {
+        final ImmutableList.Builder<ApplicationKey> applicationKeyList = new ImmutableList.Builder<ApplicationKey>().
+            add( SYSTEM_APPLICATION_KEY );
+        if ( portalRequest != null )
+        {
+            final Site site = portalRequest.getSite();
+            if ( site != null )
+            {
+                final ImmutableSet<ApplicationKey> siteApplicationKeySet = site.getSiteConfigs().getApplicationKeys();
+                applicationKeyList.addAll( siteApplicationKeySet );
+            }
+        }
+
+        final ApplicationKeys applicationKeys = ApplicationKeys.from( applicationKeyList.build() );
+        return styleDescriptorService.getByApplications( applicationKeys );
+    }
+
+    private ImageStyle getImageStyle( final Map<String, ImageStyle> imageStyleMap )
+    {
+        final String styleString = urlParams.get( STYLE_PARAM );
+        if ( styleString != null )
+        {
+            return imageStyleMap.get( styleString );
+        }
+        return null;
+    }
+
+    private String getScale( final ImageStyle imageStyle, final Integer expectedWidth )
+    {
+        final String aspectRatio = getAspectRation( imageStyle );
+
+        if ( aspectRatio != null )
+        {
+            final Matcher matcher = ASPECT_RATIO_PATTEN.matcher( aspectRatio );
+            if ( !matcher.matches() )
+            {
+                throw new IllegalArgumentException( "Invalid aspect ratio: " + aspectRatio );
+            }
+            final String horizontalProportion = matcher.group( "horizontalProportion" );
+            final String verticalProportion = matcher.group( "verticalProportion" );
+
+            final int width = Objects.requireNonNullElse( expectedWidth, DEFAULT_WIDTH );
+            final int height = width / Integer.parseInt( horizontalProportion ) * Integer.parseInt( verticalProportion );
+
+            return "block(" + width + "," + height + ")";
+        }
+
+        return expectedWidth != null ? "width(" + expectedWidth + ")" : IMAGE_SCALE;
+    }
+
+    private String getFilter( final ImageStyle imageStyle )
+    {
+        return imageStyle == null ? null : imageStyle.getFilter();
+    }
+
+    private String getAspectRation( final ImageStyle imageStyle )
+    {
+        if ( imageStyle != null )
+        {
+            final String aspectRatio = imageStyle.getAspectRatio();
+            if ( aspectRatio != null )
+            {
+                return aspectRatio;
+            }
+        }
+        return urlParams.get( SCALE_PARAM );
+    }
+
+    private Map<String, String> extractUrlParams( final String urlQuery )
+    {
+        if ( urlQuery == null )
+        {
+            return Collections.emptyMap();
+        }
+        final String query = urlQuery.startsWith( "?" ) ? urlQuery.substring( 1 ) : urlQuery;
+        return Splitter.on( '&' ).
+            trimResults().
+            withKeyValueSeparator( "=" ).
+            split( query.replace( "&amp;", "&" ) );
```

Comment:
this can't be right. It means unescaped query string can be passed. And it means some other symbols are left unescaped.